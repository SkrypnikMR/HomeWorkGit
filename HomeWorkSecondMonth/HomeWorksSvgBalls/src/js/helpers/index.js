import * as bumps from './bumps';
import * as render from './render';

export default { bumps, render};