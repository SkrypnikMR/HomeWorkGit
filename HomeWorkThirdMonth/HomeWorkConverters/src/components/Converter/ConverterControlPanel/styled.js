import styled from 'styled-components';

export const StConvertControl = styled.div`
    display: flex;
    justify-content: space-around;
    align-items: center;
    flex-direction: column;
    height: 60vh;
    width: 90%;
        p{
            font-size: 30px;
        }
`;
export const StDropDownsControl = styled.div`
    display:flex;
    justify-content: space-between;
    align-items: center;
    width: 90%;
    p{
        margin:10px;
    }
`;
