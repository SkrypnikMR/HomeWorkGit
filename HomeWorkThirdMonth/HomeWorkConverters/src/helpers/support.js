export const support = {
    lsGet: name => localStorage.getItem(name),
    lsSet: (name, item) => {
        if (typeof item === 'string') return localStorage.setItem(name, item);
        localStorage.setItem(name, JSON.stringify(item));
    },
    convert: (num, primary, secondary) => {
        const lengthSystem = {
            meters: {
                meters: 1,
                miles: 1609.344,
                yards: 1.093613,
                feet: 3.28084,
                versts: 1066.8,
            },
            miles: {
                meters: 1609.344,
                miles: 1,
                yards: 1760,
                feet: 0.003281,
                versts: 1.50857,
            },
            yards: {
                meters: 0.9144,
                miles: 0.000568182,
                yards: 1,
                feet: 3,
                versts: 0.000857143,
            },
            feet: {
                meters: 0.3048,
                miles: 0.000189394,
                yards: 0.333333,
                feet: 1,
                versts: 0.000285714,
            },
            versts: {
                meters: 1066.8,
                miles: 0.662879,
                yards: 1166.67,
                feet: 3500,
                versts: 1,
            },
        };
        return (+num * lengthSystem[primary][secondary]).toString();
    },
};
