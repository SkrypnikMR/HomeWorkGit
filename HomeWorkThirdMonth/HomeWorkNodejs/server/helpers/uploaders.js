require('dotenv/config');
const uuid = require('uuid/v4');
const AWS = require('aws-sdk');
const sharp = require('sharp');
const multer = require('multer');
const s3 = new AWS.S3({
    accessKeyId: process.env.AWS_ID,
    secretAccessKey: process.env.AWS_SECRET,
})

const fileUploader = async (req, res) => {
    const myFile = req.file.originalname.split('.');
    const fileType = myFile[myFile.length - 1];
    try {
        const params = {
            Bucket: process.env.AWS_BUCKET_NAME_FILES,
            Key: `file-${myFile[0]}-${uuid()}.${fileType}`,
            Body: req.file.buffer
        }
        const result = await s3.upload(params).promise().catch((e) => res.status(500).json({ message: 'server error' }));
        return res.status(200).json({ fileLink: result.Location });
    } catch (e) { res.status(500).json({ message: 'server error' }) }
}

const imageUploader = async (req, size, name) => {
    try {
        const myFile = req.file.originalname.split('.');
        const fileType = myFile[myFile.length - 1];

        const sizedImage = await sharp(req.file.buffer).resize({ width: size, height: size }).toBuffer();
        const params = { Bucket: process.env.AWS_BUCKET_NAME_IMAGES, Key: `${name}${myFile[0]}-${uuid()}.${fileType}`, Body: sizedImage }
        const result = await s3.upload(params).promise();

        return result.Location;
    } catch (e) {
        res.status(500).json({ message: 'server error' })
    }

}
const storage = multer.memoryStorage({ destination: (req, res, cb) => cb(null, '') });
const upload = multer({ storage, limits: { fileSize: 20971520 } }).single('file');

const corsOptions = {
    methods: ['POST', 'OPTIONS', 'GET'],
    origins: '*',
    allowedHeaders: ['Content-Type', 'multipart/form-data']
}

module.exports = { fileUploader, imageUploader, upload}