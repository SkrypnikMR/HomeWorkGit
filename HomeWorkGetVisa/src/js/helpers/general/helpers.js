import * as validate from "../validators/validators.js";
import { notifKill } from "../listeners/listeners.js";
import * as drowers from "../drowers/drowers.js";

export const click = (buttons) => {
  for (let key in buttons) {
    buttons[key].click();
  }
};

export const validator = (candidate) => {
  const vName = validate.validateName(candidate.name);
  const vBalance = validate.validateNumber(candidate.balance);
  const vAgelikeNumber = validate.validateNumber(candidate.age);
  const VAgeLength = validate.validateLength(candidate.age, 3);
  const vEnglish = validate.validateEnglish(candidate.english);
  if (vName || !vBalance || !vAgelikeNumber || VAgeLength || !vEnglish) {
    return false;
  }
  return true;
};

export const cleaner = (nodes) => {
  for (let key in nodes) {
    if (nodes[key].checked === true) {
      nodes[key].checked = false;
      continue;
    }
    nodes[key].value = "";
  }
};

export const createNotif = (str) => {
  // create notif Div
  const notif = document.createElement("div");
  const notifTextDiv = document.createElement("div");
  const notifButtonDiv = document.createElement("div");
  const notifDeleteBtn = document.createElement("button");
  const notifMessage = document.createElement("p");

  // add classes
  notif.classList.add("notification");
  notifButtonDiv.classList.add("notification__button");
  notifTextDiv.classList.add("notification__message");
  notifDeleteBtn.classList.add("button__item");
  notifDeleteBtn.classList.add("button");
  notifMessage.classList.add("message__item");

  //add textContent
  notifDeleteBtn.textContent = "X";
  notifMessage.textContent = str;

  //append chields
  notifTextDiv.appendChild(notifMessage);
  notifButtonDiv.appendChild(notifDeleteBtn);
  notif.appendChild(notifButtonDiv);
  notif.appendChild(notifTextDiv);
  document.body.appendChild(notif);

  // add event Listener
  notifDeleteBtn.addEventListener("click", notifKill.bind(null, notif));

  // autoDeath after 3s
  setTimeout(notifKill.bind(null, notif), 2000);
};

export const mutateObject = (obj) => {
  obj.balance = Number(obj.balance);
  obj.age = Number(obj.age);
  return obj;
};

export const drowInit = (candidates, canvas, context, distance_Between) => {
  drowers.drowBackground(canvas, context);
  for (let i = 0; i < candidates.length; i++) {
    const distance = distance_Between * (i + 1) + 2;
    const x = 0;
    const k = (2 * Math.PI) / 100;
    drowers.drowText(context, candidates[i].name, distance, 92, "white", 12);
    drowers.drowText(context, "balance", distance, 292, "white", 12);
    drowers.drowText(context, "documents", distance, 492, "white", 12);
    drowers.drowText(context, "age", distance, 692, "white", 12);
    drowers.drowText(context, "english", distance, 892, "white", 12);
    drowers.drowCircle(context, canvas, x, k, 100, distance, 100, 0, "green");
    candidates[i].distance = distance;
    candidates[i].balancePoint = 292;
    candidates[i].documentsPoint = 492;
    candidates[i].agePoint = 692;
    candidates[i].englishPoint = 892;
  }
};

export const sessionStorageSet = (key, value) =>
  sessionStorage.setItem(key, value);
