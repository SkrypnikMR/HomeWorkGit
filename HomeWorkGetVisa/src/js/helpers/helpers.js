import * as validate from "./validators/validators.js";
import { notifKill } from "./listeners.js";

export const click = (buttons) => {
  for (let key in buttons) {
    buttons[key].click();
  }
};

export const validator = (candidate) => {
  const vName = validate.validateName(candidate.name);
  const vBalance = validate.validateNumber(candidate.balance);
  const vAgelikeNumber = validate.validateNumber(candidate.age);
  const VAgeLength = validate.validateLength(candidate.age, 3);
  const vEnglish = validate.validateEnglish(candidate.english);
  if (vName || !vBalance || !vAgelikeNumber || VAgeLength || !vEnglish) {
    return false;
  }
  return true;
};

export const cleaner = (nodes) => {
  for (let key in nodes) {
    if (nodes[key].checked === true) {
      nodes[key].checked = false;
      continue;
    }
    nodes[key].value = "";
  }
};

export const createNotif = (str) => {
  // create notif Div
  const notif = document.createElement("div");
  const notifTextDiv = document.createElement("div");
  const notifButtonDiv = document.createElement("div");
  const notifDeleteBtn = document.createElement("button");
  const notifMessage = document.createElement("p");

  // add classes
  notif.classList.add("notification");
  notifButtonDiv.classList.add("notification__button");
  notifTextDiv.classList.add("notification__message");
  notifDeleteBtn.classList.add("button__item");
  notifDeleteBtn.classList.add("button");
  notifMessage.classList.add("message__item");

  //add textContent
  notifDeleteBtn.textContent = "X";
  notifMessage.textContent = str;

  //append chields
  notifTextDiv.appendChild(notifMessage);
  notifButtonDiv.appendChild(notifDeleteBtn);
  notif.appendChild(notifButtonDiv);
  notif.appendChild(notifTextDiv);
  document.body.appendChild(notif);

  // add event Listener
  notifDeleteBtn.addEventListener("click", notifKill.bind(null, notif));
};

export const mutateObject = (obj) => {
  obj.balance = Number(obj.balance);
  obj.age = Number(obj.age);
  return obj;
};

export const drow = (candidates, canvas, context) => {
  var box_Height = canvas.height;
  context.fillStyle = "rgba(94, 144, 252, 1)";
  context.fillRect(1, 1, canvas.width - 1, box_Height - 1);
  context.strokeRect(0.5, 0.5, canvas.width - 1, box_Height);
  const centerX = 100;
  const radius = 40;
  const distance_Between = 105;
  for (let i = 0; i < candidates.length; i++) {
    context.beginPath();
    context.font = " 12px Arial";
    context.fillStyle = "white";
    context.fillText(
      candidates[i].name,
      92 - candidates[i].name.length * 2,
      distance_Between * (i + 1) + 2
    );
    context.closePath();
    context.beginPath();
    context.font = " 12px Arial";
    context.fillStyle = "white";
    context.fillText(
      `check balance`,
      300 * 1.5,
      distance_Between * (i + 1) + 2
    );
    context.closePath();
    context.beginPath();
    context.font = " 12px Arial";
    context.fillStyle = "white";
    context.fillText(`check age`, 500 * 1.5, distance_Between * (i + 1) + 2);
    context.closePath();
    context.beginPath();
    context.font = " 12px Arial";
    context.fillStyle = "white";
    context.fillText(
      `check documents`,
      700 * 1.5,
      distance_Between * (i + 1) + 2
    );
    context.closePath();
    context.beginPath();
    context.font = " 12px Arial";
    context.fillStyle = "white";
    context.fillText(
      `check english`,
      900 * 1.5,
      distance_Between * (i + 1) + 2
    );
    context.closePath();
    let x = 0;
    let k = (2 * Math.PI) / 100;
    const kek = distance_Between * (i + 1) + 2;
    console.log(kek);
    drowCircle(context, canvas, x, k, kek);
  }
};

export function drowCircle(context, canvas, x, k, kek) {
  context.beginPath();
  context.arc(100, kek, 50, 0, x * k, false);
  context.lineWidth = 3;
  context.lineCap = "butt";
  context.strokeStyle = "green";
  context.stroke();
  context.closePath();
  let timeout;
  if (x <= 100) {
    timeout = setTimeout(() => {
      drowCircle(context, canvas, x + 1, k, kek);
    }, 50);
  } else clearTimeout(timeout);
}
