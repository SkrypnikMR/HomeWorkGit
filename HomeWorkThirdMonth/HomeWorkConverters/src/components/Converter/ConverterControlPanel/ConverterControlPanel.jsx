import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { StConvertControl, StDropDownsControl, StSpan } from './styled';
import Button from '../../UI/Button';
import Input from '../../UI/Input';
import Select from '../../UI/Select';
import { lengthOptions, currencyOptions, currencyUAH } from '../../../constants/UICONSTANTS';
import { support } from '../../../helpers/support';

class ConverterControlPanel extends Component {
    constructor(props) {
        super(props);
        this.state = {
            error: '',
        };
    }

    handleClick = () => {
        const {
            quantity,
            setConverterTablo,
            from,
            to,
            converterType,
            currency,
            setConverterQuantity } = this.props;
        const { inValid, newQuantity } = support.validateInput(quantity);
        if (inValid) return this.setState(state => ({ ...state, error: 'Only numbers' }));
        if (newQuantity === '') return this.setState(state => ({ ...state, error: 'Please enter value' }));
        const tablo = converterType === 'lengths'
            ? support.convertLengths(newQuantity, from, to)
            : support.convertCurrency(newQuantity, currency, to);
        setConverterTablo(tablo);
        setConverterQuantity('');
    }

    handleOnChange = (e) => {
        const { value } = e;
        const { setConverterQuantity } = this.props;
        setConverterQuantity(value);
    }

    componentDidMount() {
        const { getCurrency, converterType } = this.props;
        if (converterType === 'curency') getCurrency();
    }

    componentDidUpdate(prevProps) {
        const { getCurrency, converterType } = this.props;
        if (converterType === 'curency' && prevProps.converterType !== converterType) getCurrency();
    }

    render() {
        const { converterType, setConverterFrom, setConverterTo, quantity } = this.props;
        const { error } = this.state;
        return (
            <StConvertControl >
                <StDropDownsControl>
                    <Select
                        width="60%"
                        options={converterType === 'lengths' ? lengthOptions : currencyUAH}
                        from
                        to={false}
                        setConverterFrom={setConverterFrom}
                        setConverterTo={setConverterTo}
                        converterType={converterType}
                    />
                    <p>TO</p>
                    <Select
                        width="60%"
                        options={converterType === 'lengths' ? lengthOptions : currencyOptions}
                        from={false}
                        to
                        setConverterTo={setConverterTo}
                        setConverterFrom={setConverterFrom}
                        converterType={converterType}
                    />

                </StDropDownsControl>
                <Input
                    id="convertControlPanelInput"
                    placeholder="Enter your param"
                    width="90%"
                    onChange={this.handleOnChange}
                    value={quantity}
                />
                <StSpan>{error}</StSpan>
                <Button
                    content="Convert"
                    id="controllButton"
                    onClick={this.handleClick}
                />
            </StConvertControl >
        );
    }
}
ConverterControlPanel.propTypes = {
    converterType: PropTypes.string,
    setConverterFrom: PropTypes.func,
    setConverterTo: PropTypes.func,
    setConverterQuantity: PropTypes.func,
    quantity: PropTypes.string,
    setConverterTablo: PropTypes.func,
    from: PropTypes.string,
    to: PropTypes.string,
    getCurrency: PropTypes.func,
    currency: PropTypes.objectOf(PropTypes.string),
};

export default ConverterControlPanel;
