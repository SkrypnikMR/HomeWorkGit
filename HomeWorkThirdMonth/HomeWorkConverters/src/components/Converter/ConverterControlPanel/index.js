
import { connect } from 'react-redux';
import ConverterControlPanel from './ConverterControlPanel.jsx';
import { setConverterType, setConverterTo, setConverterFrom, setConverterQuantity, setConverterTablo } from '../../../store/converter/actions';

const mapStateToProps = state => (state);
const mapDispatchToProps = dispatch => ({
    setConverterType: payload => dispatch(setConverterType(payload)),
    setConverterTo: payload => dispatch(setConverterTo(payload)),
    setConverterFrom: payload => dispatch(setConverterFrom(payload)),
    setConverterQuantity: payload => dispatch(setConverterQuantity(payload)),
    setConverterTablo: payload => dispatch(setConverterTablo(payload)),
});

export default connect(mapStateToProps, mapDispatchToProps)(ConverterControlPanel);
