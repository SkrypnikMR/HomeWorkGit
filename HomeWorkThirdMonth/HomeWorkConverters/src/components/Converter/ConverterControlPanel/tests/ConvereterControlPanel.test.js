import React from 'react';
import configureStore from 'redux-mock-store';
import { shallowSmart, mountSmart } from '../../../../../__tests__/helpers/testHelper';
import ConverterControlPanel from '../ConverterControlPanel';

jest.mock('react-modal');

const mockStore = configureStore();
const store = mockStore({
    isLoading: false,
    converterType: 'lengths',
    from: '',
    to: '',
    quantity: '',
    tablo: '',
    currency: {},
    notes: [],
});

describe('Converter', () => {
    it('Should match snapshot', () => {
        const component = shallowSmart(<ConverterControlPanel
            setConverterFrom={jest.fn()}
            setConverterTo={jest.fn()}
        />, store);
        expect(component.html()).toMatchSnapshot();
    });
    it('should render StDropDownsControl', () => {
        const component = mountSmart(<ConverterControlPanel
            setConverterFrom={jest.fn()}
            setConverterTo={jest.fn()}
        />, store);
        expect(component.find('styled__StDropDownsControl')).toHaveLength(1);
    });
    it('should render MySelect', () => {
        const component = mountSmart(<ConverterControlPanel
            setConverterFrom={jest.fn()}
            setConverterTo={jest.fn()}
        />, store);
        expect(component.find('MySelect')).toHaveLength(2);
    });
    it('should render p', () => {
        const component = mountSmart(<ConverterControlPanel
            setConverterFrom={jest.fn()}
            setConverterTo={jest.fn()}
        />, store);
        expect(component.find('p')).toHaveLength(1);
    });
    it('should render Input', () => {
        const component = mountSmart(<ConverterControlPanel
            setConverterFrom={jest.fn()}
            setConverterTo={jest.fn()}
        />, store);
        expect(component.find('Input')).toHaveLength(1);
    });
    it('should render Button', () => {
        const component = mountSmart(<ConverterControlPanel
            setConverterFrom={jest.fn()}
            setConverterTo={jest.fn()}
        />, store);
        expect(component.find('Button')).toHaveLength(1);
    });
    it('should notValidInputValue Button', () => {
        const component = mountSmart(<ConverterControlPanel
            setConverterFrom={jest.fn()}
            setConverterTo={jest.fn()}
            quantity='sad'
            setConverterTablo={jest.fn()}
            from=''
            to=''
            converterType='lengths'
            currency={[]}
            setConverterQuantity={jest.fn()}
        />);
        component.find('Button').simulate('click');
        expect(component.state().error).toBe('Only numbers');
    });
    it('should empty value Button click', () => {
        const component = mountSmart(<ConverterControlPanel
            setConverterFrom={jest.fn()}
            setConverterTo={jest.fn()}
            quantity=''
            setConverterTablo={jest.fn()}
            from=''
            to=''
            converterType='lengths'
            currency={[]}
            setConverterQuantity={jest.fn()}
        />);
        component.find('Button').simulate('click');
        expect(component.state().error).toBe('Please enter value');
    });
    it('should valid value Button click converType === lengths', () => {
        const setConverterTablo = jest.fn();
        const setConverterQuantity = jest.fn();
        const component = mountSmart(<ConverterControlPanel
            setConverterFrom={jest.fn()}
            setConverterTo={jest.fn()}
            quantity='123'
            setConverterTablo={setConverterTablo}
            from='meters'
            to='yards'
            converterType='lengths'
            currency={[]}
            setConverterQuantity={setConverterQuantity}
        />);
        component.find('Button').simulate('click');
        expect(setConverterTablo).toHaveBeenCalledWith('134.51403');
        expect(setConverterQuantity).toHaveBeenCalledWith('');
    });
    it('should valid value Button click convertType === curency', () => {
        const setConverterTablo = jest.fn();
        const setConverterQuantity = jest.fn();
        const getCurrency = jest.fn();
        const component = mountSmart(<ConverterControlPanel
            setConverterFrom={jest.fn()}
            setConverterTo={jest.fn()}
            quantity='123'
            setConverterTablo={setConverterTablo}
            from='UAH'
            to='UER'
            converterType='curency'
            currency={{ UER: 3 }}
            setConverterQuantity={setConverterQuantity}
            getCurrency={getCurrency}
        />);
        component.find('Button').simulate('click');
        expect(setConverterTablo).toHaveBeenCalledWith('41');
        expect(setConverterQuantity).toHaveBeenCalledWith('');
    });
    it('should change Input Value', () => {
        const setConverterTablo = jest.fn();
        const setConverterQuantity = jest.fn();
        const component = mountSmart(<ConverterControlPanel
            setConverterFrom={jest.fn()}
            setConverterTo={jest.fn()}
            quantity='123'
            setConverterTablo={setConverterTablo}
            from='meters'
            to='yards'
            converterType='lengths'
            currency={[]}
            setConverterQuantity={setConverterQuantity}
        />);
        const testValue = 'kek';
        component.find('input').simulate('change', { target: { value: testValue } });
        expect(setConverterQuantity).toHaveBeenCalledWith(testValue);
    });
    it('should mount Element', () => {
        const setConverterTablo = jest.fn();
        const setConverterQuantity = jest.fn();
        const getCurrency = jest.fn();
        mountSmart(<ConverterControlPanel
            setConverterFrom={jest.fn()}
            setConverterTo={jest.fn()}
            quantity='123'
            setConverterTablo={setConverterTablo}
            from='meters'
            to='yards'
            converterType='curency'
            currency={[]}
            getCurrency={getCurrency}
            setConverterQuantity={setConverterQuantity}
        />);
        expect(getCurrency).toHaveBeenCalled();
    });
    it('should update component', () => {
        const setConverterTablo = jest.fn();
        const setConverterQuantity = jest.fn();
        const getCurrency = jest.fn();
        const component = mountSmart(<ConverterControlPanel
            setConverterFrom={jest.fn()}
            setConverterTo={jest.fn()}
            quantity='123'
            setConverterTablo={setConverterTablo}
            from='meters'
            to='yards'
            converterType='lengths'
            currency={[]}
            getCurrency={getCurrency}
            setConverterQuantity={setConverterQuantity}
        />);
        component.setProps({ converterType: 'curency' });
        expect(getCurrency).toHaveBeenCalled();
    });
});
