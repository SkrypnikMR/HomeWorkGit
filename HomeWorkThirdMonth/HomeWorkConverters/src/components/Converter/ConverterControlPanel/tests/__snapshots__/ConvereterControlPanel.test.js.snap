// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Converter Should match snapshot 1`] = `"<div class=\\"styled__StConvertControl-sc-19u337l-0 kJUHeb\\"><div class=\\"styled__StDropDownsControl-sc-19u337l-1 eyBSTn\\"><select width=\\"60%\\" class=\\"styled__StSelect-sc-1k7m7de-0 cJrwdK\\"><option id=\\"trueUAH\\" value=\\"uah\\" class=\\"styled__StOption-sc-1k7m7de-1 chqcFP\\">UAH</option>;</select><p>TO</p><select width=\\"60%\\" class=\\"styled__StSelect-sc-1k7m7de-0 cJrwdK\\"><option id=\\"trueEUR\\" value=\\"eur\\" class=\\"styled__StOption-sc-1k7m7de-1 chqcFP\\">EUR</option><option id=\\"trueUSD\\" value=\\"usd\\" class=\\"styled__StOption-sc-1k7m7de-1 chqcFP\\">USD</option><option id=\\"trueRUB\\" value=\\"rub\\" class=\\"styled__StOption-sc-1k7m7de-1 chqcFP\\">RUB</option>;</select></div><div width=\\"90%\\" class=\\"styled__StInputContainer-w3ykst-3 bwsXTy\\"><input type=\\"text\\" id=\\"convertControlPanelInput\\" placeholder=\\"Enter your param\\" class=\\"styled__StInput-w3ykst-1 xGBPx\\"/></div><span class=\\"styled__StSpan-sc-19u337l-2 iDQTBr\\"></span><button id=\\"controllButton\\" type=\\"button\\" class=\\"styled__StButton-sc-1a6hck8-0 cqrnfx\\">Convert</button></div>"`;
